### ISSUE TITLE
Travis build exceeds maximum time

### ISSUE THREAD

**Issue by JackMorganNZ** [uccser/cs-unplugged#572]
- https://docs.travis-ci.com/user/customizing-the-build/#Build-Timeouts
- https://docs.travis-ci.com/user/speeding-up-the-build
- https://docs.travis-ci.com/user/build-stages/
- https://docs.travis-ci.com/user/build-stages/share-docker-image/

**Comment by JackMorganNZ** [comment]
Progress started on branch `issue/572`.
Travis build showing build stages: https://travis-ci.org/uccser/cs-unplugged/builds/259968580

**Comment by ravenmaster001** [comment]
Still failing: https://travis-ci.org/uccser/cs-unplugged/builds/261248392
Note the treasure hunt builds appear to be taking a ridiculous amount of time.

**Comment by ravenmaster001** [comment]
@JackMorganNZ the deploy takes so long due to the amount of resources it is trying to build. My computer takes about 8mins for all the ones in the tests (single copy) and during deploy we ask for 30 copies. I'm thinking that resource deployment should be moved to a manual step for now.

**Comment by JackMorganNZ** [comment]
If we make it a Travis build stage, does the job time reset?

**Comment by ravenmaster001** [comment]
@JackMorganNZ Considering on my machine the command `time .\csu dev makeresources` reports back:
```
real	63m43.671s
user	0m0.196s
sys	0m0.044s
```
I'm not sure that it's going to matter.

**Comment by JackMorganNZ** [comment]
Well that is an issue... Having 30 copies per resources ideally the minimum as that is a standard class sizes (though when I first asked educators, their answer was 100 copies!). Once we have the render system up and running this step would be skipped.

I might try one last solution instead of switching to manual deployment, but if it fails, manual will have to do!

**Comment by JackMorganNZ** [comment]
Here are the times for generating resources on my machine:

```
Arrows                 : 2 seconds
Barcode Checksum Poster: 2 seconds
Binary Cards           : 23 seconds
Binary Cards (Small)   : 15 seconds
Binary to Alphabet     : 1 second
Binary Windows         : 18 seconds
Grid                   : 1 second
Job Badges             : 1 second
Left and Right Cards   : 1 seconds
Modulo Clock           : 2 seconds
Parity Cards           : 3 seconds
Piano Keys             : 3 seconds
Searching Cards        : 34 seconds
Sorting Network        : 47 seconds
Sorting Network Cards  : 14 seconds
Train Stations         : 1 second
Treasure Hunt          : 4139 seconds (69 minutes)
```

**Comment by JackMorganNZ** [comment]
Solved in 1a7dc80 by splitting deployment into several stages.
